[IMP] mail_notification_history: make configurable from UI

mkoeck opened the pull request:
This pull request updates the way the "Mail Notification With History" feature is enabled and managed in Odoo, shifting from a code-based configuration to a user-friendly setting in the UI. It also introduces additional translations and updates related documentation and tests.

**Configuration and Feature Activation:**
- The mechanism to enable mail history in notifications is now managed via a new `include_mail_history` Boolean field on the `ir.model` model, configurable through the Odoo UI (Settings → Technical → Database Structure → Models), instead of a Python class variable (`_mail_notification_include_history`). [[1]](diffhunk://#diff-74182fc23df385eeacd7654aa117f43710bfabf0cbbccd3ef7881132a38d6ea9R1-R6) [[2]](diffhunk://#diff-1881cfd6ab9cac0f66501552a905e41f5824e7f292b77b47e02f257073a12c07R1-R12) [[3]](diffhunk://#diff-e0f00324d29e099cb076a73d24176e225b5339068512d5761c54697802e8dfe6L12-R17) [[4]](diffhunk://#diff-e5895a55149bb5a41f4d6abfa0e6da128d8e0f657bba43b5569666e4e628800eL1-L10)

**Documentation Updates:**
- All documentation and installation instructions have been updated to reflect the new UI-based activation method, replacing references to the old class variable approach. [[1]](diffhunk://#diff-3e8d2da2b550926ddaa1b71e1ba899b6e290d02ad631194833c22b5e51ab7b2dL45-R47) [[2]](diffhunk://#diff-32dd021e183bef22cd1dce0f27eb99eb329d4c929c035f4dc3856f4e3db404f3L2-R2) [[3]](diffhunk://#diff-9eec9218745303c4496d36d6fd7e3a89af7a6bd2bf6b3ec307977108070c2d48L394-R396)

**Testing Adjustments:**
- Tests have been refactored to set the new `include_mail_history` field on models, instead of patching the class variable, to reflect the new configuration approach. [[1]](diffhunk://#diff-a61599472ffa0feba1aa6e4bc038d49726bf582f490c2054902ed7704540d743L29-R27) [[2]](diffhunk://#diff-a61599472ffa0feba1aa6e4bc038d49726bf582f490c2054902ed7704540d743L80-R73)

**Translation and Internationalization:**
- Added a German translation file (`de.po`) and updated the translation template (`mail_notification_with_history.pot`) to include new field labels and help texts for the new feature. [[1]](diffhunk://#diff-a4c3bef5185564675dec9a03c2f5214f49b6f0f17f32a48371202c5f8d5ebf2fR1-R48) [[2]](diffhunk://#diff-85c16e8b8a143d28e909de198b55b272a28796ccb1cc9311932b426c99c039c7L7-R10) [[3]](diffhunk://#diff-85c16e8b8a143d28e909de198b55b272a28796ccb1cc9311932b426c99c039c7L22-R44)

**Module Manifest and File Structure:**
- Updated the module manifest to include the new view for the `ir.model` form and removed obsolete code and imports related to the previous activation method. [[1]](diffhunk://#diff-a25383f863be1302f9f5e14aa6f917fee4b0bdb4b627ded5d8ac242274b9e1e8L14-R14) [[2]](diffhunk://#diff-4e0c255e30f4e1197ee23bea3dc6b72b91b36d80b17e75f55da8ab3637ff838eL2-R2) [[3]](diffhunk://#diff-e5895a55149bb5a41f4d6abfa0e6da128d8e0f657bba43b5569666e4e628800eL1-L10) [[4]](diffhunk://#diff-a61599472ffa0feba1aa6e4bc038d49726bf582f490c2054902ed7704540d743L4-L5)

mkoeck committed:
[IMP] mail_notification_with_history: replace class variable with config field on ir.model

mkoeck committed:
[FIX] mail_notification_with_history: correct imports

mkoeck committed:
[i18n] mail_notification_with_history: add new translations

mkoeck committed:
[i18n] mail_notification_with_history: add german translation

mkoeck committed:
[i18n] mail_notification_with_history: fix german translation

mkoeck committed:
[FIX] mail_notification_with_history: run pre-commit

TDu commented:
Good thing that it would be configurable through the UI.
But this change, without migration script, will break the module for the one already using it :exclamation:

mkoeck committed:
[IMP] mail_notification_with_history: add migration script

mkoeck committed:
[IMP] mail_notification_with_history: add deprecation warning

mkoeck commented:
Hi @TDu thank you for your quick response. I added a migration script that should fix this I think. Unfortunately it requires the upgrade util - I did not find a better way to do it without it. I guess if the dependency is not wanted, I could also copy paste the env function from the upgrade utils into the upgrade script.

Also added a deprecation warning should the attribute still be set on a class. What do you think?

mkoeck committed:
[FIX] mail_notification_with_history: run pre-commit

mkoeck committed:
[FIX] mail_notification_with_history: add line breaks for E501

mkoeck committed:
[FIX] mail_notification_with_history: pre-commit

mkoeck commented:
I just reran the pre-commit and I see that it removed the upgrade-utils from the requirements.txt. Is that to be expected?

TDu commented:
@OCA/social-maintainers could you reopen this one, please.
